scripts_regression_tests fails not captured

FAIL: test_self_build_cprnc (CIME.tests.test_sys_cime_case.TestCimeCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/glade/work/jedwards/sandboxes/cesm2_x_alpha/cime/scripts/lib/CIME/tests/test_sys_cime_case.py", line 721, in test_self_build_cprnc
    self._wait_for_tests(self._baseline_name, always_wait=True)
  File "/glade/work/jedwards/sandboxes/cesm2_x_alpha/cime/scripts/lib/CIME/tests/base.py", line 262, in _wait_for_tests
    self.run_cmd_assert_result(
  File "/glade/work/jedwards/sandboxes/cesm2_x_alpha/cime/scripts/lib/CIME/tests/base.py", line 137, in run_cmd_assert_result
    self.assertEqual(stat, expected_stat, msg=msg)
AssertionError: 100 != 0 : 
    COMMAND: /glade/work/jedwards/sandboxes/cesm2_x_alpha/cime/scripts/Tools/wait_for_tests  *fake_testing_only_20220103_081933/TestStatus
    FROM_DIR: /glade/scratch/jedwards/scripts_regression_test.20220103_081933
    SHOULD HAVE WORKED, INSTEAD GOT STAT 100
    OUTPUT: FAIL ERS_Ln7.f19_g16_rx1.A.cheyenne_intel (phase RUN)
    Case dir: ERS_Ln7.f19_g16_rx1.A.cheyenne_intel.fake_testing_only_20220103_081933
    ERRPUT: 
    

----------------------------------------------------------------------
Ran 1 test in 385.827s

FAILED (failures=1)
All pass, removing directory: /glade/scratch/jedwards/scripts_regression_test.20220103_081933

--- a/scripts/lib/CIME/tests/scripts_regression_tests.py
+++ b/scripts/lib/CIME/tests/scripts_regression_tests.py
@@ -50,6 +50,8 @@ from CIME.tests.base import BaseTestCase
 
 os.environ["CIME_GLOBAL_WALLTIME"] = "0:05:00"
 
+TEST_RESULT = None
+
 
 def write_provenance_info(machine, test_compiler, test_mpilib, test_root):
     curr_commit = get_current_commit(repo=CIMEROOT)
@@ -67,7 +69,11 @@ def write_provenance_info(machine, test_compiler, test_mpilib, test_root):
 
 
 def cleanup(test_root):
-    if os.path.exists(test_root):
+    if (
+        os.path.exists(test_root)
+        and TEST_RESULT is not None
+        and TEST_RESULT.wasSuccessful()
+    ):
         testreporter = os.path.join(test_root, "testreporter")
         files = os.listdir(test_root)
         if len(files) == 1 and os.path.isfile(testreporter):
@@ -266,6 +272,8 @@ OR
         test_suite = unittest.defaultTestLoader.loadTestsFromNames(ns.tests)
 
     test_runner = unittest.TextTestRunner(verbosity=2)
+
+    global TEST_RESULT
 
     TEST_RESULT = test_runner.run(test_suite)
 

